network/src/protocols/mod.rs

```diff
@@ -348,4 +352,17 @@ impl CKBProtocolContext for DefaultCKBProtocolContext {
     fn protocol_id(&self) -> ProtocolId {
         self.proto_id
     }
+
+    fn inbound_eviction_condition(&self) -> bool {
+        let status = self.network_state.connection_status();
```

Comment:
check bootnode mode first:

```suggestion
        if self.network_state.config.bootnode_mode {
           ...
        }
```